Framework/Core/include/Framework/ArrowTypes.h

```diff
@@ -77,21 +77,26 @@ struct arrow_array_for<double[N]> {
   using type = arrow::FixedSizeListArray;
   using value_type = double;
 };
-template <>
-struct arrow_array_for<std::vector<int>> {
-  using type = arrow::ListArray;
-  using value_type = int;
-};
-template <>
-struct arrow_array_for<std::vector<float>> {
-  using type = arrow::ListArray;
-  using value_type = float;
-};
-template <>
-struct arrow_array_for<std::vector<double>> {
-  using type = arrow::ListArray;
-  using value_type = double;
-};
+
+#define ARROW_VECTOR_FOR(_type_)                \
+  template <>                                   \
+  struct arrow_array_for<std::vector<_type_>> { \
+    using type = arrow::ListArray;              \
+    using value_type = _type_;                  \
+  };
+
+ARROW_VECTOR_FOR(uint8_t);
+ARROW_VECTOR_FOR(uint16_t);
+ARROW_VECTOR_FOR(uint32_t);
+ARROW_VECTOR_FOR(uint64_t);
```

Comment:
I think this might need to use the ROOT type. Otherwise  we end up with the usual problem that uint64_t is different types on different platform...

Reply:
That is handled by TableBuilder, using stuff in `RootTableBuilderHelpers.h`. It will do remapping when needed.